zebra-chain/src/primitives/address.rs

```diff
@@ -185,10 +163,34 @@ impl Address {
             Self::Transparent(address) => Some(address.to_string()),
             Self::Sapling { address, network } => {
                 let data = address.to_bytes();
-                let address = ZcashAddress::from_sapling(network.into(), data);
+                let address = ZcashAddress::from_sapling(network.to_zcash_address(), data);
                 Some(address.encode())
             }
             Self::Unified { .. } => None,
         }
     }
 }
+
+impl NetworkKind {
+    /// Converts a [`zcash_address::Network`] to a [`NetworkKind`].
+    ///
+    /// This method is meant to be used for decoding Zcash addresses in zebra-rpc methods.
+    fn from_zcash_address(network: zcash_address::Network) -> Self {
```

Comment:
It was to keep it private to this module and so it would be easier to search for where it's used to avoid using it outside of RPC methods, but since it's okay for now to use `Testnet` HRPs for custom Testnets, I'll go with the `From` trait.

Reply:
That makes sense, feel free to keep it as it is. I was just wondering what the reason was.